src/manager/node.c

```diff
@@ -260,53 +276,97 @@ static int node_match_unit_properties_changed(sd_bus_message *m, void *userdata,
         }
 
         const UnitSubscriptionsKey key = { (char *) unit };
-        UnitSubscriptions *usubs = NULL;
-
-        usubs = hashmap_get(node->unit_subscriptions, &key);
-        if (usubs == NULL) {
-                return 0;
+        UnitSubscriptions *usubs = hashmap_get(node->unit_subscriptions, &key);
+        if (usubs != NULL) {
+                UnitSubscription *usub = NULL;
+                UnitSubscription *next_usub = NULL;
+                LIST_FOREACH_SAFE(subs, usub, next_usub, usubs->subs) {
+                        Subscription *sub = usub->sub;
+                        int r = sub->handle_unit_property_changed(sub->monitor, node->name, unit, interface, m);
```

Comment:
This is not your fault, but I think there is another issue here. What if a single monitor has multiple subscriptions that overlap. We will then send multiple events to this single monitor.

I wonder if we should have the two loops just compute a vector of of (unique) monitors, and then loop over those and call handle_unit_property_changed.

Reply:
I moved the `handle_xyz` functions from the `Subscription` to the `Monitor`, which seems to be a better fit. When I tried to compute a unique list of monitors with only a hashmap (using the monitor id as key), that resulted in segfaults due to the `handle_xyz` functions being NULL - since the hashmap copies everything. I decided to use a hashmap just for the ids to track which monitor is already added to a list of unique monitors. 
This also results in the `unit_subscription_contains_monitor` to be removed.